### PR TITLE
Add sensor ams5915

### PR DESCRIPTION
**Pull request by borob-engineering**
Hello,
I would like to add the implementation for the pressure sensor ams5915 of [analog microelectronics](https://www.analog-micro.com/products/pressure-sensors/board-mount-pressure-sensors/ams5915). It is tested using a nucleo-g474re.

Thanks

### CODE REVIEW

File: dts/bindings/vendor-prefixes.txt [hunk 1]
```diff
@@ -55,6 +55,7 @@ ams	AMS AG
 amstaos	AMS-Taos Inc.
 analogix	Analogix Semiconductor, Inc.
 andestech	Andes Technology Corporation
+analogmicro	analog microelectronics GmbH
```
Comment (kartben) [inline dts/bindings/vendor-prefixes.txt:58]: needs to be alphabetically sorted. Also please capitalize `Analog Microelectronics` and consider fixing typo in associated commit

File: dts/bindings/sensor/anamicro,ams5915.yaml
Comment (kartben) [inline dts/bindings/sensor/anamicro,ams5915.yaml:1]: file name needs to be consistant with vendor prefix

File: drivers/sensor/ams5915/Kconfig [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# AMS5915 pressure and temperature sensor configuration options
+
+# Copyright (c) 2024 BoRob Engineering
+#
+# SPDX-License-Identifier: Apache-2.0
+
+config AMS5915
+	bool "AMS 5915 is a series of high precision, digital pressure sensors"
```
Comment (kartben) [inline drivers/sensor/ams5915/Kconfig:8]:
```suggestion
	bool "AMS5915 I2C-based pressure and temperature sensor"
```

File: samples/sensor/ams5915/README.rst
Comment (kartben) [inline samples/sensor/ams5915/README.rst:1]: please drop the introduction of this new sample. It is not needed as the sensor driver doesn't have any vendor-specific API, and people can use any existing generic sample to exercise it.

**Review by kartben — CHANGES_REQUESTED** [review]
thanks for the PR! Some early feedback from having a quick look

File: samples/sensor/ams5915/README.rst
Comment (borob-engineering) [reply samples/sensor/ams5915/README.rst:1]: Just to be sure: should I drop the whole sample folder of asm5915 or just the Readme file?

**Pull request closed without merging**